dev-infrastructure/Makefile

```diff
@@ -138,6 +138,13 @@ create-mock-identities:
 	SUBSCRIPTION_ID=$(shell az account show --query id --output tsv) \
 	./scripts/create-sp-for-rbac.sh
 
+	APPLICATION_NAME=aro-dev-msi-mock2 \
+	KEY_VAULT_NAME=aro-hcp-dev-svc-kv \
```

Comment:
Is this the right keyvault?

Reply:
This has been copied from the previous role assignment section, where the same identity(`aro-dev-msi-mock2`) is being assigned a role.

Reply:
I am trying to add role assignment, that is assigning role `Azure Red Hat OpenShift KMS Plugin - Dev` to identity `aro-dev-msi-mock2` at subscription level. 
Here keyvault is provided because `create-sp-for-rbac.sh` needs it to store app credential. 

I want to do this because `aro-dev-msi-mock2` is used by hypershift in personal dev environment to communicate with the keyvault(Not the keyvault which it provided here). I want this to test etcd encryption of hosted cluster. 

While we can proceed with a manual role assignment for now(To avoid credential reset), this change is intended to ensure it’s maintained through code going forward.